stl/inc/ranges

```diff
@@ -6821,6 +6821,682 @@ namespace ranges {
+            } else {
+                return _Iterator<false>{_Zip::_Tuple_transform(_RANGES end, _Ranges)};
+            }
+        }
+
+        _NODISCARD constexpr auto end() const noexcept(_Is_end_noexcept<true, const _ViewTypes...>) // strengthened
+            requires(range<const _ViewTypes>&&...) {
+            if constexpr (!_Zip::_Is_common<const _ViewTypes...>) {
+                return _Sentinel<true>{_Zip::_Tuple_transform(_RANGES end, _Ranges)};
+            } else if constexpr ((random_access_range<const _ViewTypes> && ...)) {
+                return _RANGES next(begin(), iter_difference_t<_Iterator<true>>{_Get_signed_size()});
+            } else {
+                return _Iterator<true>{_Zip::_Tuple_transform(_RANGES end, _Ranges)};
+            }
+        }
+
+        _NODISCARD constexpr auto size() noexcept(
+            noexcept(_Zip::_Tuple_for_each(_RANGES size, _Ranges))) // strengthened
+            requires(sized_range<_ViewTypes>&&...) {
+            using _Common_size_type          = common_type_t<range_size_t<_ViewTypes>...>;
+            using _Common_unsigned_size_type = _Make_unsigned_like_t<_Common_size_type>;
+
+            auto _Smallest_view_size_value = (numeric_limits<_Common_unsigned_size_type>::max)();
+
+            _Zip::_Tuple_for_each(
+                [&_Smallest_view_size_value]<class _View>(_View& _Curr_view) {
+                    const auto _Curr_view_size_value =
+                        static_cast<_Common_unsigned_size_type>(_RANGES size(_Curr_view));
+
+                    if (_Curr_view_size_value < _Smallest_view_size_value)
+                        _Smallest_view_size_value = _Curr_view_size_value;
+                },
+                _Ranges);
+
+            return _Smallest_view_size_value;
+        }
+
+        _NODISCARD constexpr auto size() const
+            noexcept(noexcept(_Zip::_Tuple_for_each(_RANGES size, _Ranges))) // strengthened
+            requires(sized_range<const _ViewTypes>&&...) {
+            using _Common_size_type          = common_type_t<range_size_t<_ViewTypes>...>;
+            using _Common_unsigned_size_type = _Make_unsigned_like_t<_Common_size_type>;
+
+            auto _Smallest_view_size_value = (numeric_limits<_Common_unsigned_size_type>::max)();
+
+            _Zip::_Tuple_for_each(
+                [&_Smallest_view_size_value]<class _View>(const _View& _Curr_view) {
+                    const auto _Curr_view_size_value =
+                        static_cast<_Common_unsigned_size_type>(_RANGES size(_Curr_view));
+
+                    if (_Curr_view_size_value < _Smallest_view_size_value)
+                        _Smallest_view_size_value = _Curr_view_size_value;
+                },
+                _Ranges);
+
+            return _Smallest_view_size_value;
+        }
+
+    private:
+        _NODISCARD constexpr auto _Get_signed_size() noexcept(noexcept(size())) {

```

Comment:
I believe that was done to avoid narrowing conversions in `zip_view::size()` when constructing `iter_difference_t` instances. I must've had a brain aneurysm that day, though, because we could also just do `static_cast<iter_difference_t<...>>(size())`.

I'll be removing that function.